Dynamic input expiration rounds up to full minutes (feature request: disable rounding)

Hi,

While using the dynamic input expiration logic, I noticed that the expiration time is always rounded up to full minutes. In my use case I need the expiration to be calculated **exactly based on seconds**, not rounded.

The current implementation in `inputs/dynamic.go` does the following:

```go
minutes := int(math.Ceil(float64(totalSeconds) / 60.0))
expiresAt := time.Now().Add(time.Duration(minutes) * time.Minute)
```

This means durations are always rounded up:

* `30s` → `1 minute`
* `61s` → `2 minutes`

For my use case this causes the expiration to last longer than intended.

**Local fix**

I changed the function to use the exact number of seconds:

```go
expiresAt := time.Now().Add(time.Duration(totalSeconds) * time.Second)
```

Complete function:

```go
func (d *DynamicInput) calculateDynamicExpiration(duration string) time.Time {
        totalSeconds, ok := utils.ParseDurationToSeconds(duration)
        if !ok {
                return d.handleUnsupportedFormat(duration)
        }

        if totalSeconds <= 0 {
                slog.Error("Duration must be greater than 0 seconds - will expire immediately", "input", d.GetName(), "duration", duration)
                return time.Now()
        }

        expiresAt := time.Now().Add(time.Duration(totalSeconds) * time.Second)

        slog.Debug(
                "Calculated dynamic expiration",
                "input", d.GetName(),
                "duration", duration,
                "totalSeconds", totalSeconds,
                "expiresAt", expiresAt.Format("15:04:05"),
        )

        return expiresAt
}

```

This works correctly and produces precise expiration times.

After making this change I initially got the following build error:

```
$ go build
# zwfm-metadata/inputs
inputs/dynamic.go:6:2: "math" imported and not used
```

Removing the `math` import fixed the issue and the project builds successfully.

**Feature request**

Since modifying the source code means I need to reapply the change after every `git pull`, it would be great if this behavior could be configurable.

For example, a configuration option for dynamic inputs such as:

```
exact_expiration: true
```

or

```
round_to_minutes: false
```

So users can choose between:

* the current behavior (rounded minutes)
* exact expiration based on seconds

This would make the feature more flexible without breaking existing setups.

Thanks for considering!

--- a/README.md
+++ b/README.md
@@ -119,7 +119,7 @@ HTTP API for live updates
 - `secret` (optional) - Authentication secret for API calls
 - `expiration.type` - `"dynamic"` (expires based on song duration), `"fixed"` (expires after a set number of minutes), or `"none"` (never expires)
 - `expiration.minutes` (required if type=fixed, optional for type=dynamic) - Number of minutes until expiration. When `type` is `"dynamic"`, this serves as a fallback when the duration parameter is missing or invalid
-- `expiration.roundUpMinutes` (optional, default: true) - When `true` (or omitted), dynamic expiration rounds up to full minutes (e.g., 3:30 → 4 minutes). This prevents metadata "flapping" when short segments like talk or jingles follow a song. Set to `false` to use exact second-based expiration
+- `expiration.roundUpMinutes` (optional, default: true, only for type=dynamic) - When `true` (or omitted), dynamic expiration rounds up to full minutes (e.g., 3:30 → 4 minutes). This prevents metadata "flapping" when short segments like talk or jingles follow a song. Set to `false` to use exact second-based expiration
 
 #### API Usage
 ```bash

--- a/config-example.json
+++ b/config-example.json
@@ -26,7 +26,8 @@
       "settings": {
         "secret": "supersecret123",
         "expiration": {
-          "type": "dynamic"
+          "type": "dynamic",
+          "roundUpMinutes": true
         }
       }
     },

--- a/config/config.go
+++ b/config/config.go
@@ -55,7 +55,7 @@ type DynamicInputConfig struct {
 	Expiration struct {
 		Type           string `json:"type"`                     // "dynamic", "fixed", "none"
 		Minutes        int    `json:"minutes,omitempty"`        // Fallback minutes for dynamic, or fixed duration
-		RoundUpMinutes *bool  `json:"roundUpMinutes,omitempty"` // Round up expiration to full minutes (default: true)
+		RoundUpMinutes *bool  `json:"roundUpMinutes,omitempty"` // Round up dynamic expiration to full minutes (default: true)
 	} `json:"expiration"`
 }
 

--- a/inputs/dynamic.go
+++ b/inputs/dynamic.go
@@ -20,6 +20,10 @@ type DynamicInput struct {
 
 // NewDynamicInput initializes an HTTP API-driven input with the given settings.
 func NewDynamicInput(name string, settings config.DynamicInputConfig) *DynamicInput {
+	if settings.Expiration.RoundUpMinutes != nil && settings.Expiration.Type != "dynamic" {
+		slog.Warn("roundUpMinutes is only used with expiration type \"dynamic\" and will be ignored", "input", name, "type", settings.Expiration.Type)
+	}
+
 	return &DynamicInput{
 		InputBase:      core.NewInputBase(name),
 		settings:       settings,

